src/components/CreatePostForm.js

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from "react";
+import { v4 as uuid } from "uuid";
+import { useNavigate, useParams } from "react-router-dom";
+import {
+  getPostsFromLocalStorage,
+  savePostsToLocalStorage,
+} from "../utils/localStorage";
+import "../styles/createPost.css";
+import { blogThemesAvailable, readerGroups } from "../constants";
+
+const CreatePostForm = () => {
+  const navigate = useNavigate();
+  const { postId } = useParams();
+
+  const getInitialFormData = () => ({
+    id: uuid(),
+    title: "",
+    author: "",
+    themes: [],
+    readerGroup: "",
+    content: "",
+    image: "",
+    likes: 0,
+  });
+
+  const getPostData = (postId) => {
+    const posts = getPostsFromLocalStorage();
+    const post = posts.find((p) => p.id === postId) || {};
+    return { ...getInitialFormData(), ...post };
+  };
+
+  const isEditMode = !!postId;
+  const initialFormData = isEditMode
+    ? getPostData(postId)
+    : getInitialFormData();
+
+  const [formData, setFormData] = useState(initialFormData);
+  const [errors, setErrors] = useState({});
+
+  const validateForm = () => {
+    const newErrors = {};
+
+    if (!formData.title.trim()) {
+      newErrors.title = "Title is required";
+    }
+
+    if (!formData.author) {
+      newErrors.author = "Author is required";
+    }
+
+    if (formData.themes.length === 0) {
+      newErrors.theme = "At least one theme must be selected";
+    }
+
+    if (!formData.readerGroup) {
+      newErrors.readerGroup = "Reader Group is required";
+    }
+
+    if (!formData.content.trim()) {
+      newErrors.content = "Content is required";
+    }
+
+    setErrors(newErrors);
+
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const handleChange = (e) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleCheckboxChange = (e) => {
+    const theme = e.target.value;
+    const updatedThemes = formData.themes.includes(theme)
+      ? formData.themes.filter((t) => t !== theme)
+      : [...formData.themes, theme];
+
+    setFormData({
+      ...formData,
+      themes: updatedThemes,
+    });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const posts = getPostsFromLocalStorage();
+
+    const FormData = isEditMode
+      ? {
+          ...formData,
+          updatedAt: new Date().toISOString(),
+        }
+      : {
+          ...formData,
+          createdAt: new Date().toISOString(),
+        };
+
+    if (validateForm()) {
+      const updatedPosts = isEditMode
+        ? posts.map((post) => (post.id === postId ? FormData : post))
+        : [...posts, FormData];
+      savePostsToLocalStorage(updatedPosts);
+
+      navigate("/");
```

Comment:
first validate form and then edit the formdata.

Reply:
done